src/Components/Components/src/Forms/FieldIdentifier.cs

```diff
@@ -0,0 +1,111 @@
+            // Note that we do allow an empty string. This is used by some validation systems
+            // as a place to store object-level (not per-property) messages.
+            FieldName = fieldName ?? throw new ArgumentNullException(nameof(fieldName));
+        }
+
+        /// <summary>
+        /// Gets the object that owns the editable field.
+        /// </summary>
+        public object Model { get; }
+
+        /// <summary>
+        /// Gets the name of the editable field.
+        /// </summary>
+        public string FieldName { get; }
+
+        /// <inheritdoc />
+        public override int GetHashCode()
+            => (Model, FieldName).GetHashCode();
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+            => obj is FieldIdentifier otherIdentifier
+            && otherIdentifier.Model == Model
+            && string.Equals(otherIdentifier.FieldName, FieldName, StringComparison.Ordinal);
+
+        private static void ParseAccessor<T>(Expression<Func<T>> accessor, out object model, out string fieldName)
+        {
+            var accessorBody = accessor.Body;
+
+            // Unwrap casts to object
+            if (accessorBody is UnaryExpression unaryExpression
+                && unaryExpression.NodeType == ExpressionType.Convert
+                && unaryExpression.Type == typeof(object))
+            {
+                accessorBody = unaryExpression.Operand;
+            }
+
+            if (!(accessorBody is MemberExpression memberExpression))
+            {
+                throw new ArgumentException("The accessor is not supported because its body is not a MemberExpression");
+            }
+
+            // Identify the field name. We don't mind whether it's a property or field, or even something else.
+            fieldName = memberExpression.Member.Name;
+
+            // Get a reference to the model object
+            // i.e., given an value like "(something).MemberName", determine the runtime value of "(something)",
+            switch (memberExpression.Expression)
+            {
+                case ConstantExpression constantExpression:
+                    model = constantExpression.Value;
+                    break;
+                case MemberExpression nestedMemberExpression:
+                    // It would be great to cache this somehow, but it's unclear there's a reasonable way to do
+                    // so, given that it embeds captured values such as "this". We could consider special-casing
+                    // for "() => something.Member" and building a cache keyed by "something.GetType()" with values
+                    // of type Func<object, object> so we can cheaply map from "something" to "something.Member".
+                    var modelLambda = Expression.Lambda(nestedMemberExpression);
+                    var modelLambdaCompiled = (Func<object>)modelLambda.Compile();
+                    model = modelLambdaCompiled();
+                    break;
```

Comment:
You could also do this using reflection directly which would probably be faster and safer. It's possible that `.Compile()` causes unbounded memory growth, but we also use it in MVC so #YOLO

Reply:
Sounds like a good future enhancement. I'd rather not try to predict all the use cases for the first preview.